homeassistant/components/light/hue.py

```diff
@@ -48,254 +43,188 @@
     'Color temperature light': SUPPORT_HUE_COLOR_TEMP
     }
 
-ATTR_EMULATED_HUE_HIDDEN = 'emulated_hue_hidden'
 ATTR_IS_HUE_GROUP = 'is_hue_group'
 
-# Legacy configuration, will be removed in 0.60
-CONF_ALLOW_UNREACHABLE = 'allow_unreachable'
-DEFAULT_ALLOW_UNREACHABLE = False
-CONF_ALLOW_IN_EMULATED_HUE = 'allow_in_emulated_hue'
-DEFAULT_ALLOW_IN_EMULATED_HUE = True
-CONF_ALLOW_HUE_GROUPS = 'allow_hue_groups'
-DEFAULT_ALLOW_HUE_GROUPS = True
-
-PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend({
-    vol.Optional(CONF_HOST): cv.string,
-    vol.Optional(CONF_ALLOW_UNREACHABLE): cv.boolean,
-    vol.Optional(CONF_FILENAME): cv.string,
-    vol.Optional(CONF_ALLOW_IN_EMULATED_HUE): cv.boolean,
-    vol.Optional(CONF_ALLOW_HUE_GROUPS,
-                 default=DEFAULT_ALLOW_HUE_GROUPS): cv.boolean,
-})
-
-MIGRATION_ID = 'light_hue_config_migration'
-MIGRATION_TITLE = 'Philips Hue Configuration Migration'
-MIGRATION_INSTRUCTIONS = """
-Configuration for the Philips Hue component has changed; action required.
-
-You have configured at least one bridge:
-
-    hue:
-{config}
-
-This configuration is deprecated, please check the
-[Hue component](https://home-assistant.io/components/hue/) page for more
-information.
-"""
-
-SIGNAL_CALLBACK = 'hue_light_callback_{}_{}'
-
-
-def setup_platform(hass, config, add_devices, discovery_info=None):
-    """Set up the Hue lights."""
-    if discovery_info is None or 'bridge_id' not in discovery_info:
-        return
-
-    if config is not None and config:
-        # Legacy configuration, will be removed in 0.60
-        config_str = yaml.dump([config])
-        # Indent so it renders in a fixed-width font
-        config_str = re.sub('(?m)^', '      ', config_str)
-        hass.components.persistent_notification.async_create(
-            MIGRATION_INSTRUCTIONS.format(config=config_str),
-            title=MIGRATION_TITLE,
-            notification_id=MIGRATION_ID)
-
-    bridge_id = discovery_info['bridge_id']
-    bridge = hass.data[hue.DOMAIN][bridge_id]
-    unthrottled_update_lights(hass, bridge, add_devices)
 
+class BoolProxy:
+    """Proxy value for a boolean."""
```

Comment:
I think this requires a longer explanation. What does this solve?

Reply:
Good point. I'll remove it and just pass bridge object through like the old implementation.

Reply:
What it was solving btw was that we're sharing a boolean value with all the light objects and are able to change it without having to update each light object individually.

Reply:
:+1: